src/tests/tracing/eventpipe/applystartuphook/ApplyStartupHookValidation.cs

```diff
@@ -64,6 +66,90 @@ public static async Task<bool> TEST_ApplyStartupHookAtStartupSuspension()
+
+                        SessionConfiguration config = new(
+                            circularBufferSizeMB: 1000,
+                            format: EventPipeSerializationFormat.NetTrace,
+                            providers: new List<Provider> { 
+                                new Provider(AppEventSource.SourceName, 0, EventLevel.Verbose)
+                            });
+
+                        Logger.logger.Log("Starting EventPipeSession over standard connection");
+                        using Stream eventStream = EventPipeClient.CollectTracing(pid, config, out ulong sessionId);
+                        Logger.logger.Log($"Started EventPipeSession over standard connection with session id: 0x{sessionId:X}");
+
+                        using EventPipeEventSource source = new(eventStream);
+                        TaskCompletionSource completionSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+                        source.Dynamic.All += (TraceEvent traceEvent) =>
+                        {
+                            if (AppEventSource.SourceName.Equals(traceEvent.ProviderName) && nameof(AppEventSource.Running).Equals(traceEvent.EventName))
+                                completionSource.TrySetResult();
+                        };
+
+                        _ = Task.Run(() => source.Process());
+
+                        Logger.logger.Log($"Send ResumeRuntime Diagnostics IPC Command");
+                        // send ResumeRuntime command (0x04=ProcessCommandSet, 0x01=ResumeRuntime commandid)
+                        var message = new IpcMessage(0x04,0x01);
+                        Logger.logger.Log($"Sent: {message.ToString()}");
+                        IpcMessage response = IpcClient.SendMessage(stream, message);
+                        Logger.logger.Log($"received: {response.ToString()}");
+                        CheckResponse(response, ref fSuccess);
+                        
+                        Logger.logger.Log("Start waiting for any managed event.");
+                        await completionSource.Task.ConfigureAwait(false);
+
+                        Logger.logger.Log("Stopping trace.");
+                        EventPipeClient.StopTracing(pid, sessionId);
+                    }
+
+                    Logger.logger.Log("Waiting to accept diagnostic connection.");
+                    using (Stream stream = await server.AcceptAsync())
+                    {
+                        Logger.logger.Log("Accepted diagnostic connection.");
+
+                        IpcAdvertise advertise = IpcAdvertise.Parse(stream);
+                        Logger.logger.Log($"IpcAdvertise: {advertise}");
+
+                        string startupHookPath = Hook.Basic.AssemblyPath;
+                        Logger.logger.Log($"Send ApplyStartupHook Diagnostic IPC: {startupHookPath}");
+                        IpcMessage message = CreateApplyStartupHookMessage(startupHookPath);
```

Comment:
Something you might feel free to punt: There's a copy of the client in src/tests/tracing/eventpipe/common/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj. Since you are adding the tests herem do you want to add the methods to the client and that gives you the roundtrip for the changes you have in diagnostics?

Reply:
The `ReverseServer` class used in these tests are incompatible with the copy of the client library. I've added the necessary methods in a prior PR, but we aren't really using them. I can try to take a look at trying to use the client library separately hopefully with few modifications.

Reply:
Thanks, didn't know